test/journal_test.go

```diff
@@ -89,6 +89,92 @@ func TestJournalCrSimple(t *testing.T) {
 	)
 }
 
+// bob creates a conflicting file while running the journal, and then
+// hits another conflict while trying to put the resolution.
+// Regression test for KBFS-1590.
+func TestJournalCrConflictOnPut(t *testing.T) {
```

Comment:
I'm not sure if this test is worth having -- it's pretty complex and brittle.  The panic I added to `folderBranchOps` might be good enough.  What do you think?